webui/src/pages/extension-detail/extension-detail.tsx

```diff
@@ -165,10 +165,12 @@ export class ExtensionDetailComponent extends React.Component<ExtensionDetailCom
         }
 
         return <React.Fragment>
-            <Box className={this.props.classes.head} style={{
-                backgroundColor: extension.galleryColor,
-                color: extension.galleryTheme === 'dark' ? '#ffffff' : undefined
-            }} >
+            <Box className={this.props.classes.head}
+                style={{
+                    backgroundColor: this.props.prefersDarkMode ? undefined : extension.galleryColor,
+                    color: extension.galleryTheme === 'dark' ? '#ffffff' : undefined
+                }}
+            >
```

Comment:
`extension.galleryColor` shall have priority here.

We have to make sure that when `extension.galleryTheme` is undefined, we default to the value passed through `props` (or `props.pageSettings`).

Reply:
For example, the line
```typescript
const themeClass = extension.galleryTheme === 'dark'
        ? this.props.classes.darkTheme : this.props.classes.lightTheme;
```
in `renderHeader` could become
```typescript
const themeClass = (extension.galleryTheme || this.props.pageSettings.themeType) === 'dark' ?
        this.props.classes.darkTheme : this.props.classes.lightTheme;
```

Reply:
> `extension.galleryColor` shall have priority here, please keep the previous value of `backgroundColor`.

@spoenemann  If we prioritize `extension.galleryColor` this is what we get when the theme is dark:

![image](https://user-images.githubusercontent.com/46004116/80583242-67eb7d80-8a29-11ea-96e8-d32f67757251.png)

![image](https://user-images.githubusercontent.com/46004116/80583355-86ea0f80-8a29-11ea-9ce9-5f4b9689e30b.png)

that's why I have made it unprioritized `extension.galleryColor` when the theme is dark.

Reply:
You also need to respect `extension.galleryTheme` as I explained above. When that is set to `'light'`, the header must be rendered with a dark font, and vice versa.

Reply:
Thanks! @spoenemann I have made the change there is a small problem though:

![image](https://user-images.githubusercontent.com/46004116/80651678-c64b4700-8a8f-11ea-9d03-a1a98353b5e2.png)


On dark Mode where the `extension.galleryTheme` is `light` the horizontal `<hr />` does not change it's colour to black.

![image](https://user-images.githubusercontent.com/46004116/80651769-f397f500-8a8f-11ea-966a-be7a515c83cc.png)

Reply:
The `TextDivider` component has a `theme` parameter that controls its color.

Reply:
Thanks! the problem is now fixed.